### DISCUSSION TITLE
Calculation of CO2FluidPropertiesTest

### DISCUSSION THREAD

**Discussion by shin-chan-cpu** [idaholab/moose#26951] [Q&A General]
When I use the CO2FluidPropertiesTest to calculate the single-phase calculation, the input card is written as follows
```
[FluidProperties]
  [fp]
    type = CO2FluidPropertiesTest
  []
[]
[FunctorMaterials]
  [fluid_props_to_mat_props]
    type = GeneralFunctorFluidProps
    fp = fp
    pressure = 'pressure'
    T_fluid = 'T_fluid'
    speed = 'velocity_norm'
    characteristic_length = 2
    porosity = 0
  []
  [ins_fv]
    type = INSFVEnthalpyFunctorMaterial
    temperature = 'T_fluid'
    rho = ${rho}
  []
[]
```
But after the last run, an error is reported.
```
Time Step 0, time = 0

Time Step 1, time = 0.001, dt = 0.001


*** ERROR ***
The following error occurred in the object "fp", of type "CO2FluidProperties".

fp: virtual void SinglePhaseFluidProperties::rho_from_p_T(const DualReal&, const DualReal&, DualReal&, DualReal&, DualReal&) const derivative derivatives not implemented.
```

I looked in the header file is rho_from_p_T this function, why he still error？

**Comment by GiudGiud** [comment]
Hello

The signature with the ADReal for every component and derivatives of the density has not been implemented. 

You ll want to use a parameter on the GeneralFunctorFluidProps to neglect the derivatives of the time derivative of the density. It s already there, I can check the exact name later

Guillaume

**Reply by GiudGiud** [reply]
The name of the parameter is: `neglect_derivatives_of_density_time_derivative`

**Reply by shin-chan-cpu** [reply]
Sorry, even after I changed this parameter to true, he still reported the same error

**Reply by shin-chan-cpu** [reply]
virtual void SinglePhaseFluidProperties::e_from_p_rho(libMesh::Real, libMesh::Real, libMesh::Real&, libMesh::Real&, libMesh::Real&) const derivatives not implemented.
This is the mistake he reported later.

**Reply by GiudGiud** [reply]
That's a different error but yes still an error.
This means you need to implement this routine to make this fluid property work for your application

**Reply by shin-chan-cpu** [reply]
thank you very much
